block-producer/src/state_view/mod.rs

```diff
@@ -121,24 +121,23 @@ where
         &self,
         block: Block,
     ) -> Result<(), ApplyBlockError> {
-        self.store.apply_block(block.clone()).await?;
+        let account_ids_in_block: Vec<AccountId> =
+            block.updated_accounts.iter().map(|(account_id, _, _)| *account_id).collect();
+        let produced_nullifiers = block.produced_nullifiers.clone();
+
+        self.store.apply_block(block).await?;
```

Comment:
I assume that `store.apply_block` encodes the `Block` into a protobuf message. Meaning it doesn't need ownership of `Block`. If that is the case, I think we should change it to receive a reference, and then here we don't need the copies.

Reply:
The linked code is creating new objects. At first sight it seems none of them require ownership.